nrf52-hal-common/src/spim.rs

```diff
@@ -56,6 +57,71 @@ impl_spim_ext!(
 /// - The over-read character is hardcoded to `0`.
 pub struct Spim<T>(T);
 
+impl<T> embedded_hal::blocking::spi::Transfer<u8> for Spim<T> where T: SpimExt
+{
+   type Error = Error;
+
+    fn transfer<'w>(&mut self, words: &'w mut [u8]) -> Result<&'w [u8], Error> {
+        let mut offset:usize = 0;
+        while offset < words.len() {
+            let datalen = min(EASY_DMA_SIZE, words.len() - offset);
+            let dataptr = offset + (words.as_ptr() as usize);
+            offset += EASY_DMA_SIZE;
+
+            self.do_spi_dma_transfer(dataptr as u32,datalen as u32,dataptr as u32,datalen as u32,|_|{})?;
+
+            // Conservative compiler fence to prevent optimizations that do not
+            // take in to account DMA
+            compiler_fence(SeqCst);
+        }
+        Ok(words)
+    }
+}
+impl<T> embedded_hal::blocking::spi::Write<u8> for Spim<T> where T: SpimExt
+{
+   type Error = Error;
+
+    fn write<'w>(&mut self, words: &'w [u8]) -> Result<(), Error> {
+        let p = words.as_ptr() as usize;
+        // Mask on segment where Data RAM is located on nrf52840 and nrf52832
+        // Upper limit is choosen to entire area where DataRam can be placed
+        if SRAM_LOWER <= p && p < SRAM_UPPER {
+            let mut offset:usize = 0;
+            while offset < words.len() {
+                let datalen = min(EASY_DMA_SIZE, words.len()  - offset);
+                let dataptr = offset + (words.as_ptr() as usize);
+                offset += EASY_DMA_SIZE;
```

Comment:
Same suggestion as above, feel free to ignore.

```suggestion
                offset += datalen;
```